✨ feat(chat): Chat 앱 초기 설정 및 코드 스타일 정리

BE12-OZ opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #123
- 작업 요약: chat 앱의 초기 설정을 완료하고, 프로젝트 전체의 모델링 기반을 강화

## 📄 상세 내용
- [x] Core/models.py UUID, Soft Delete, TDD 기능 구현 -> develop 브랜치에 있는 데이터 기준으로 수정 (진행 전 협의 필수)
   
- [x] Chat 앱 초기 설정 및 모델링 (apps/chat/)
- 앱 생성 및 구조화: chat 앱을 생성하고, users 및 app_notifications 등 다른 앱과의 일관성을 위해 models를 디렉토리로 분리하는 등 구조 리팩토링
- 모델 정의: ERD와 API 명세서에 따라 ChatMessage와 LastReadMessage 모델을 정의
- 아직 존재하지 않는 users, studies 앱의 모델은 문자열 참조 방식으로 안전하게 처리
- ERD와 API 명세서 간의 file_url 필드 불일치를 발견하고, API 명세서에 따라 모델에 필드를 추가 (협의 필요)
- 서비스 레이어 도입: apps/chat/services.py를 생성하여, 추후 비즈니스 로직을 views에서 분리할 수 있는 기반을 마련
- 기존에 verbose_name 관련 코드 삭제 (관리자 페이지 및 전체적으로 가독성을 높이기 위해 적용 했으나 영어로 통일하기로 한 것을 뒤늦게 파악하여 즉각 수정)

- [x] 코드 품질 관리
- 포매팅 적용: black과 isort를 사용하여, 이번에 변경된 모든 코드(core, chat 앱)의 스타일을 프로젝트 표준에 맞게 정리

## 📄 2차 수정 내역
- [x] develop 브랜치 병합 및 충돌 해결:
- 팀과의 협업 및 현재 작업 단계를 고려하여, 우선 `develop` 브랜치의 `BaseModel`을 기준으로 코드를 통일
- 두 브랜치의 앱이 모두 포함되도록 `INSTALLED_APPS` 목록을 병합

- [x] CI 오류 수정:
- core.models가 BaseModel로 변경됨에 따라, TimeStampModel을 상속받고 있던 chat 앱의 모델들이 BaseModel을 상속받도록 수정 (ImportError 해결)
- core 앱의 테스트 코드(test_models.py)가 더 이상 존재하지 않는 TimeStampModel을 참조하던 문제를 해결 (삭제)

- [x] 코드 스타일 통일:
- isort, black을 실행하여 프로젝트 전체의 코드 포매팅을 최신 상태로 유지
- chat 앱 모델에 포함되어 있던 verbose_name 속성을 제거하여 코드 스타일을 통일

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항
- 현재 PR에는 마이그레이션 파일이 미포함
- 사유: chat 모델이 아직 develop 브랜치에 병합되지 않은 usstudies 앱에 의존하고 있기 때문에, 현재 상태에서는 마이그레이션 파일 생성이 불가능
- 해결 계획: users, studies 등 선행 PR이 develop에 병합된 후, develop의 최신 코드를 가져와 마이그레이션을 생성하여 별도의 PR 또는 추가 커밋으로 올릴 예정

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

BE12-OZ committed:
✨ feat(chat): Chat 앱 초기 설정 및 Core 기반 모델 강화

- apps/chat 앱 생성 및 INSTALLED_APPS 등록

- ERD 및 API 명세서 기반 ChatMessage, LastReadMessage 모델 정의

- apps/core/models.py에 TimeStampModel, UUIDTimeStampModel 기반 모델 생성

- Soft-delete 기능 및 SoftDeleteManager 구현

- core 기본 모델에 대한 단위 테스트 코드 작성 (apps/core/tests/test_models.py)

2joonkim commented on apps/chat/models/chat_message.py:
```diff
@@ -0,0 +1,31 @@
+        related_name="chat_messages",
+        verbose_name="스터디 그룹",
+    )
+    content = models.TextField(verbose_name="메시지 내용")
```
content varchar [not null] // 채팅 내용

이부분 텍스트필드가 아닌데 텍스트필드로 넣으신 이유가 있으실까요? Erd상과 일치시켜주세요!!

BE12-OZ replied:
ERD와의 일관성에 대해 인지했습니다 ! 리뷰 감사합니다. 
content 필드에 대해 TextField를 사용하게 된건  사용자가 긴 설명을 채팅에 붙여넣는 경우가 있지 않을까 해서 찾아보고 적용하게 되었습니다. 
말씀하신대로 ERD 상의 varchar로 변경하겠습니다.

2joonkim commented on apps/chat/models/chat_message.py:
```diff
@@ -0,0 +1,31 @@
+        verbose_name="스터디 그룹",
+    )
+    content = models.TextField(verbose_name="메시지 내용")
+    file_url = models.URLField(max_length=255, null=True, blank=True, verbose_name="첨부 파일 URL")
```
이건 무슨 부분일까요?

BE12-OZ replied:
이 부분은 PR 설명에 있듯이 ERD와 API 명세서 간의 불일치를 발견하고, API 명세서에 따라 추가했지만 협의가 필요한 상황입니다. 
예를 들어 채팅방에서 PDF 자료라던지 이미지 등 공유하는 상황이 발생할 걸 염두해서 추가한 부분인데, 기능을 포함하기로 결정 된다면 ERD 업데이트가 필요하고, 포함하지 않는다면 이 필드는 바로 삭제하는게 맞는 것 같습니다.

2joonkim replied:
위에서 말씀드렸듯이 우선순위를 erd 상으로만 진행해주세요! 향 후에 따로 코치님과 상의하여 진행하시면 좋을 거 같습니다

BE12-OZ replied:
넵! 알겠습니다 ! 말씀하신 부분 수정하겠습니다.

BE12-OZ replied:
@2joonkim  조교님, 리뷰 주셔서 감사합니다! 
말씀해주신 대로 content 필드를 CharField로 수정하고 file_url 필드를 삭제하여 다시 한번 ERD와 일관성을 맞추었습니다. 
다시 한번 확인 부탁드립니다!

BE12-OZ committed:
refactor(chat): 리뷰 피드백 반영하여 ChatMessage 모델 수정

BE12-OZ committed:
fix(chat): core.BaseModel을 상속하도록 모델 수정

BE12-OZ committed:
fix: develop 브랜치 병합, 충돌 및 CI 에러 해결

choismjames23 commented on apps/chat/models/chat_message.py:
```diff
@@ -0,0 +1,26 @@
+
+class ChatMessage(BaseModel):
+    sender = models.ForeignKey(
+        "users.User",
```
유저 왜래키로 불러올때 settings.AUTH_USER_MODEL 로 불러오는게 좋아보입니다

BE12-OZ replied:
앗 ! 방금 확인했습니다. 
바로 수정하겠습니다 ! 감사합니다.

choismjames23 commented on apps/chat/models/last_read_message.py:
```diff
@@ -0,0 +1,29 @@
+    #     related_name="last_read_messages",
+    # )
+    user = models.ForeignKey(
+        "users.User",
```
여기도요!

BE12-OZ committed:
refactor(chat): settings.AUTH_USER_MODEL 사용하도록 수정

Meoyoug reviewed (commented):
StudyGroup 모델 먼저 머지하고 해당 모델에대한 FK 생성 반영해서 머지할게요!

BE12-OZ committed:
fix(core): Align models.py with develop branch

BE12-OZ commented:
현재 docker hub 에러로 이미지 재빌드 불가(30분 경과) -> 일정 시간 지난 후 진행 예정

Meoyoug commented:
> 현재 docker hub 에러로 이미지 재빌드 불가(30분 경과) -> 일정 시간 지난 후 진행 예정

CI 에서는 블랙터지고있고 Test는 도커쪽 복구된것같아서 다시실행후 해결된 것 같아요

BE12-OZ committed:
🎨 style: black 포맷터를 사용한 코드 스타일 수정

- black 포맷터를 실행하여 chat 앱의 코드 스타일을 수정합니다.

BE12-OZ committed:
🎨 style(core): black 포맷터를 사용한 코드 스타일 수정

- CI 오류 해결을 위해 black 포맷터를 실행하여 core 앱의 코드 스타일을 수정합니다.

Meoyoug commented on apps/chat/services.py:
스터디 그룹 모델 추가되어있으니 주석해제하고 추가해주세요

BE12-OZ committed:
fix(chat): apps/chat/admin.py 파일 삭제

코드 리뷰 지침에 따라 apps/chat/admin.py 파일을 삭제합니다.
해당 파일은 chat 앱의 관리자 페이지 등록을 위해 추가되었으나,
공통 컴포넌트 및 다른 앱 파일은 건드리지 말라는 지침에 따라 삭제 조치합니다.

Meoyoug commented on apps/chat/admin.py:
저희 장고어드민 사용할거아니고 별도의 어드민 페이지 제작해서 어드민 API 붙여야합니다 이파일은 제거부탁드려요

BE12-OZ committed:
SCRUM-12 fix(chat): mypy import 경로 수정

apps/chat/services.py에서 StudyGroup import 경로를 'apps.studies.models.groups'로 변경하여 mypy 오류 해결.
